backend/app/notify/email.go

```diff
@@ -0,0 +1,156 @@
+package notify
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	log "github.com/go-pkgz/lgr"
+	"github.com/pkg/errors"
+	"gopkg.in/gomail.v2"
+)
+
+// EmailParams contain settings for email set up
+type EmailParams struct {
+	Server    string
+	Port      int
+	From      string
+	Username  string
+	Password  string
+	KeepAlive time.Duration
+}
+
+// Email implements notify.Destination for email
+type Email struct {
+	server    string
+	port      int
+	from      string
+	username  string
+	password  string
+	keepAlive time.Duration
+	sendChan  chan *gomail.Message
+	errChan   chan error
+}
+
+const emailConnectionKeepAlive = 30 * time.Second
+
+//NewEmail makes email object for notifications and run sending daemon
+func NewEmail(params EmailParams) (*Email, error) {
+
+	res := Email{
+		server:    params.Server,
+		port:      params.Port,
+		from:      params.From,
+		username:  params.Username,
+		password:  params.Password,
+		keepAlive: params.KeepAlive,
+		sendChan:  make(chan *gomail.Message),
+		errChan:   make(chan error),
+	}
+
+	if res.keepAlive == 0 {
+		res.keepAlive = emailConnectionKeepAlive
+	}
+
+	log.Printf("[DEBUG] create new email notifier for server %s with user %s, keepalive=%s",
+		res.server, res.username, res.keepAlive)
+
+	// Test connection before starting a daemon.
+	tmpConn, err := gomail.NewDialer(res.server, res.port, res.username, res.password).Dial()
+	if err != nil {
+		return &res, errors.Wrapf(err, "error establishing test connecting to '%s':%d with username '%s'",
+			res.server, res.port, res.username)
+	}
+	if err = tmpConn.Close(); err != nil {
+		return &res, errors.Wrapf(err, "error closing test connection to %s:%d",
+			res.server, res.port)
+	}
+
+	// Activate server goroutine.
+	go res.activate()
+	// TODO: server goroutine never dies! This exiter should be run once first time we get context
+	// Close the channel to stop the mail daemon
+	// go func() { <-ctx.Done(); close(res.sendChan) }()
+
+	return &res, nil
+}
+
+// Send email from request to address in settings
+func (e *Email) Send(ctx context.Context, req request) error {
+	log.Printf("[DEBUG] send notification via %s, comment id %s", e, req.comment.ID)
+
+	messageBody := prepareBody(req)
+
+	// Create message.
+	m := gomail.NewMessage()
+	m.SetHeader("From", e.from)
+	// TODO: figure out where "to" addresses come from
+	//m.SetAddressHeader("To", req.Address, req.Name)
```

Comment:
I have a question here on the scope of the changes: where should I get this `to` field from? Should I create a separate API method on subscribing to some change which will write email->post\author relationship into a database, so then on new comment `notify` will check it and generate multiple email events? (Last part can be altered to have database query and multiple send logic inside `email`, not on higher `notify` level). Or email module just need to send whatever comes to it, same as telegram counterpart for now?